debian/postinst

```diff
@@ -27,6 +27,38 @@ SYSTEMD_WANTS_AUTO_ATTACH_LINK="/etc/systemd/system/multi-user.target.wants/ua-a
 SYSTEMD_HELPER_ENABLED_AUTO_ATTACH_DSH="/var/lib/systemd/deb-systemd-helper-enabled/ua-auto-attach.service.dsh-also"
 SYSTEMD_HELPER_ENABLED_WANTS_LINK="/var/lib/systemd/deb-systemd-helper-enabled/multi-user.target.wants/ua-auto-attach.service"
 
+update_files_based_on_contract_delta() {
+  prev_release=$1
+  curr_release=$2
+
+    python3 -c "
+from uaclient.config import UAConfig
+from uaclient.contract import process_entitlements_delta
+
+past_entitlements=UAConfig(series='$prev_release').entitlements
+new_entitlements=UAConfig(series='$curr_release').entitlements
+
+process_entitlements_delta(
+   past_entitlements=past_entitlements,
+   new_entitlements=new_entitlements,
+   allow_enable=True
+)
+"
+}
+
+get_release_from_pkg_version(){
+    pkg_version=$1
+
+    if [[ $pkg_version =~ "14.04" ]]; then
+        echo "trusty"
+    elif [[ $pkg_version =~ "16.04" ]]; then
+        echo "xenial"
+    elif [[ $pkg_version =~ "18.04" ]]; then
+        echo "bionic"
+    elif [[ $pkg_version =~ "20.04" ]]; then
+        echo "focal"
+    fi
```

Comment:
Could probably do this in a case statement (which makes me feel "safer" that string comparisons in POSIX shell as we already have examples of case/esac matching). We also should have a catch all if we don't find a matching known version.

```suggestion
    case "$pkg_version" in
         *14.04*) echo focal;;
         *16.04*) echo xenial;;
         *18.04*) echo bionic;;
         *20.04*) echo focal;;
         *) echo "unknown-previous-release";;
    esac
```